### ISSUE TITLE
Rename file to match class name results in exceptions

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#53336]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Rename-file-to-match-class-name-results-/1415420)._

---
Trying to execute the "rename file to {class name}.cs" results in exceptions if a file with such a name already exists but is not part of the project.
The same when renaming class using the Rename refactoring - VS tries to update the file name automatically resulting in similar exceptions.

---
#### Feedback Bot on 5/6/2021, 11:13 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Feedback Bot on 5/11/2021, 02:11 PM: 

<p>Thank you for sharing your feedback! Our teams prioritize action on product issues with broad customer impact. See details at: <a target='_blank' href="https://docs.microsoft.com/en-us/visualstudio/ide/report-a-problem?view=vs-2019#faq">https://docs.microsoft.com/en-us/visualstudio/ide/report-a-problem?view=vs-2019#faq</a>. In case you need answers to common questions or need assisted support, be sure to use <a target='_blank' href="https://visualstudio.microsoft.com/vs/support/">https://visualstudio.microsoft.com/vs/support/</a>. We’ll keep you posted on any updates to this feedback.</p>

---
### Original Solutions
(no solutions)

**Comment by ryzngard** [comment]
Seems like a fairly straightforward problem that only files in solution are considered for unique name generation. @jasonmalinowski what do you think? If so, might be worth marking as help wanted for good first issue.

**Comment by jasonmalinowski** [comment]
@ryzngard I admit this may be somewhat tricky for a first issue; refactorings aren't usually doing something like File.Exists to see if the potential file is already there. It wouldn't be the end of the world but would probably result in some eyebrows raised during code review.

**Comment by ryzngard** [comment]
@jasonmalinowski that's fair, does this need to go through design review at all? `File.Exists` would make this _better_ but not infallible. I understand the hesitation, but since this explicitly is a file operation I don't see too much controversy in trying to behave better.

**Comment by jasonmalinowski** [comment]
@ryzngard: yeah, probably would be worth a design review. There's also (theoretically) bugs with the File.Exists since if they delete the file they won't have a way to refresh the diagnostic. But that's probably a better bug to have than this one.

**Comment by jmarolf** [comment]
## Design Review

**Description:** 
 - Doing file IO such as a rename is fraught with difficulty. Is there a general approach that we as a team take when handling these issues?
**Conclusion:**
 - `File.Exists` will solve some of the issues, as long as it is done in a codefix/refactoring and not and analyzer this  fine
